lib/user_app.dart

```diff
@@ -160,13 +156,15 @@ class UserApp extends StatelessWidget {
                                   );
                                 case '/swap_page':
                                   return FadeInRoute(
-                                    builder: (_) => const SwapPage(),
+                                    builder: (_) => BlocProvider(
+                                      create: (BuildContext context) => SwapInBloc(ServiceInjector().breezLib),
+                                      child: const SwapPage(),
+                                    ),
```

Comment:
We add BlocProviders under [MultiBlocProvider](https://github.com/breez/c-breez/blob/ae9bf51a9865df394269c17507d4a7f42360ee11/lib/main.dart#L52) on top level.

Reply:
This is a different kind of bloc that needs to be bound to the page lifecycle (closed when the page is disposed) so the best practice is to wrap the widget that needs it, see here: https://pub.dev/documentation/flutter_bloc/latest/flutter_bloc/Bloc/close.html